### PR TITLE
Backup Optim : Primary Key instead of Limit Offset

### PR DESCRIPTION
**Pull request by GautierBoudeweel**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Please be specific when describing this change. What did you change? Why?
| Type?             | bug fix / improvement / new feature / refactor
| BC breaks?        | yes / no
| Deprecations?     | yes / no
| Fixed ticket?     | Fixes #{issue URL here}, Fixes #{another issue URL here}
| Sponsor company   | Your company or customer's name goes here (if applicable).
| How to test?      | Indicate how to verify that this change works as expected.

### CODE REVIEW

File: classes/Task/Upgrade/BackupDb.php [hunk 1]
```diff
@@ -219,7 +221,11 @@ public function run(): int
             if (!in_array($table, $ignore_stats_table)) {
                 do {
                     $backup_loop_limit = $this->container->getState()->getBackupLoopLimit();
-                    $data = $this->container->getDb()->executeS('SELECT * FROM `' . $table . '` LIMIT ' . (int) $backup_loop_limit . ',200', false, false);
+                    if ($primary_key) {
+                        $data = $this->container->getDb()->executeS('SELECT * FROM `' . $table . '` WHERE `' . $primary_key . '` BETWEEN ' . (int) $backup_loop_limit . ' AND ' . (int) $backup_loop_limit + 200 , false, false);
```
Comment (Quetzacoalt91) [inline classes/Task/Upgrade/BackupDb.php:225]: Note that in this implementation, if there is a gap in the rows that is bigger than a batch size, nothing would be returned by this request and the sync of the backup would be considered as done.

Comment (Quetzacoalt91) [reply classes/Task/Upgrade/BackupDb.php:225]: That's a great idea, and this should be stored in a cache. We would avoid reloading the same request at each HTTP request.

**Comment by GautierBoudeweel** [comment]
There is another issue with composite primary key, like in ps_cart_rule_combination
We could ignore the primary_key optimization for these tables with composite indexes, but it is precisely on these tables that we find the largest volumes and that the need for optimization is the greatest

**Comment by ga-devfront** [comment]
I'm going to close this one backup SQL optimization already done on this one https://github.com/PrestaShop/autoupgrade/pull/894.

**Pull request closed without merging**